3_javascript/javascript1/week2/javascript.js

```diff
@@ -0,0 +1,146 @@
+//Welcome to my second experience in JAVASCRIPT :)
+
+/* console.log("*******     Flight booking fullname function    *******");
+ function getFullname(firstname, surname) {
+     return `${firstname} ${surname}`;
+}
+
+ const fullname1 = getFullname("Helena", "Hashemi");
+ const fullname2 = getFullname("Shahram", "Shokri");
+
+ console.log(fullname1);
+ console.log(fullname2); */
+
+console.log("*******     Flight booking fullname function / ('Formal fullnam')     *******");
+function getFullname(firstname, surname, gender, userFormalName) {
+
+    if(gender === "male") {
+    return `Lord ${firstname} ${surname}`;
+}
+if(gender === "female"){
+    return `Lady ${firstname} ${surname}`;
+}
+if (gender === ""){
+    return `${firstname} ${surname}`;
+}
+}
+
+const userFormalName = getFullname("Soeren", "Sepehr", "");
+const userFormalName1 = getFullname("Helena", "HME", "female");
+const userFormalName2 = getFullname("Shahram", "S.K.Loo", "male");
+const userFormalName3 = getFullname("Carl-Emil", "R.H.Hansen", "male" );
+
+console.log(userFormalName);
+console.log(userFormalName1);
+console.log(userFormalName2);
+console.log(userFormalName3);
+
+
+console.log("*******     Event application     *******");
+function getEventWeekDay(howManyDaysUntilEvent) {
+    const weekdays = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    const todayDate = new Date();
+    let dayIndex = todayDate.getDay();
+    const eventIndex = (dayIndex + howManyDaysUntilEvent) % 7;
+
+    return weekdays[eventIndex];
+}
+
+console.log(getEventWeekDay(5));
+console.log(getEventWeekDay(11));
+
+//A comment for dear mentor:
+/* The question was very complex.
+Despite my extensive efforts to understand it
+(searching on Google, checking educational websites, watching videos on YouTube,
+looking at others' answers, and seeking help from ChatGPT)
+I still found it challenging to grasp the issue.
+I tried my best to comprehend it but wasn't very successful.
+Please explain the solution to this problem in a simple manner. */
+
+console.log("*******     Weather wear     *******");
+// function outsideTempreture(tempreture) {
+//     const wearInFrazing = "jacket and gloves";
+//     const wearInCold = "sweater";
+//     const wearInWarm = "top and shorts";
+
+//     if(tempreture <= 10) {
+//         return `Frazing...! Wear ${wearInFrazing}.`
+//     }
+//     if(tempreture > 10 && tempreture < 19) {
+//         return `Almost cold...! Wear ${wearInCold}.`
+//     } else {
+//         return `Sun shine...! Wear ${wearInWarm} and remember suncream and sunglasses.`
+//     }
+// }
+// const wearInWarm = outsideTempreture(27);
+// console.log(wearInWarm);
```

Comment:
This function that is commented out is the rught answer!

Reply:
I tried to show you how I think about answering the question, then I started to add the answers by commenting one of them.